swig/cpp/src/Libyang.cpp

```diff
@@ -29,107 +29,96 @@ extern "C" {
 #include "tree_schema.h"
 }
 
-using namespace std;
-
-Context::Context(ly_ctx *ctx, S_Deleter deleter) {
-    _ctx = ctx;
-    _deleter = deleter;
-}
+Context::Context(ly_ctx *ctx, S_Deleter deleter):
+    ctx(ctx),
+    deleter(deleter)
+{};
 Context::Context(const char *search_dir, int options) {
-    _ctx = ly_ctx_new(search_dir, options);
-    if (NULL == _ctx) {
-        throw runtime_error("can not create new context");
+    ctx = ly_ctx_new(search_dir, options);
+    if (nullptr == ctx) {
+        throw std::runtime_error("can not create new context");
     }
-    _deleter = S_Deleter(new Deleter(_ctx));
+    deleter = S_Deleter(new Deleter(ctx));
```

Comment:
nit: I noticed that (unlike how it was last year in sysrepo) there's no conditional typedef for `S_Something`; it is never a `Something *`, it's always a `std::shared_ptr<Something>`. If that is the case, please consider switching to `std::make_shared`. It's more idiomatic C++, and it saves one memory allocation.

Reply:
Ok, this will take some time to change.